src/app-layout/index.tsx

```diff
@@ -266,19 +265,33 @@ const OldAppLayout = React.forwardRef(
       }
     );
 
-    const defaultDrawerSize = selectedDrawer?.size || toolsWidth;
+    const getSizes = () => {
+      const sizes: { [id: string]: number } = {};
+      if (!drawers) {
+        return {};
+      }
 
-    const [drawerSize = defaultDrawerSize, setDrawerSize] = useControllable(
-      selectedDrawer?.size,
+      for (const item of drawers.items) {
+        if (item.resizable && item.size) {
+          sizes[item.id] = item.size;
+        }
+      }
+      return sizes;
+    };
+
+    const [drawerSizes = {}, setDrawerSizes] = useControllable(
+      Object.keys(getSizes()).length > 0 ? getSizes() : undefined,
```

Comment:
non-blocking: Since we call the `getSizes` function twice (and always), we can do a tiny performance improvement by calling the function immediately when it's defined:

```js
const sizes = (() => {
      const sizes: { [id: string]: number } = {};
      if (!drawers) {
        return {};
      }

      // etc ...

      return sizes;
})();